uk.co.compucorp.civicrm.hrleaveandabsences/hrleaveandabsences.php

```diff
@@ -714,3 +714,25 @@ function _hrleaveandabsences_set_has_leave_approved_by_as_default_relationship_t
     );
   }
 }
+
+/**
+ * Implementation of the hook_civicrm_validateForm.
+ *
+ * @param string $formName
+ * @param array $fields
+ * @param array $files
+ * @param object $form
+ * @param array $errors
+ */
+function hrleaveandabsences_civicrm_validateForm($formName, &$fields, &$files, &$form, &$errors) {
+  if ($formName == 'CRM_Contact_Form_Relationship') {
+    //True If the contact is being assigned a relationship with itself
+    if($fields['related_contact_id'] == $form->_contactId) {
```

Comment:
Instead of adding a comment, please extract this to a separate function with a better name describing what it does. My suggestion for the name is '_hrleaveandabsences_contact_is_being_assigned_as_its_own_leave_approver'.

This function will then look something like this:
```php
if (_hrleaveandabsences_contact_is_being_assigned_as_its_own_leave_approver($form)) {
  $errors['relationship_type_id'] = ts('You cannot assign a contact as its own leave approver');
}
```